src/pokemonClass.ts

```diff
@@ -0,0 +1,69 @@
+export type pokemonType =
+  | "grass"
+  | "poison"
+  | "fire"
+  | "water"
+  | "bug"
+  | "normal"
+  | "flying"
+  | "electric"
+  | "ground"
+  | "fairy"
+  | "psychic"
+  | "fighting"
+  | "rock";
+
+export abstract class Pokemon {
+  constructor() {}
```

Comment:
constructor should probably take all the fields: name, maxHP (and set current hp equal to max hp), image, attack, defense, types, moves